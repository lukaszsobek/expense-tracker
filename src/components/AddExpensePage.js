import React from "react"
import { connect } from "react-redux"
import { ExpenseForm } from "./"

import { addExpense } from "../actions"

export class AddExpensePage  extends React.Component {

    onSubmit = expense => {
        this.props.addExpense(expense)
        this.props.history.push("/")
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="app-container">
                        Add Expense
                    </div>
                </div>

                <ExpenseForm
                        onSubmit={this.onSubmit}
                    />
            </div>
        )
    }
}


const mapDispatchToProps = dispatch => ({
    addExpense: expense => dispatch(addExpense(expense))
})

export default connect(undefined,mapDispatchToProps)(AddExpensePage)