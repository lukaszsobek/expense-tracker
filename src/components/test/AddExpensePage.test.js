import React from "react"
import { shallow } from "enzyme"

import { AddExpensePage } from "../AddExpensePage"

const testExpense = {
    description: "testDesc",
    note: "testNote",
    amount: 200,
    createdAt: 0,
}

describe("AddExpensePage",() => {
    it("renders component",() => {
        const wrapper = shallow(<AddExpensePage />)
        expect(wrapper).toMatchSnapshot()
    })

    it("submits component",() => {
        const addExpenseSpy = jest.fn()
        const historySpy = {
            push: jest.fn()
        }
        const wrapper = shallow(<AddExpensePage 
            addExpense={addExpenseSpy}
            history={historySpy}
            />)
        wrapper.find("ExpenseForm").prop("onSubmit")(testExpense)
        expect(addExpenseSpy).toHaveBeenCalledWith(testExpense)
        expect(historySpy.push).toHaveBeenCalledWith("/")
    })
})
