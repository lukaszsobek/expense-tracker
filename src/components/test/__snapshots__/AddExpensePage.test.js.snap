// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddExpensePage renders component 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="app-container"
    >
      Add Expense
    </div>
  </div>
  <div
    className="app-container"
  >
    <ExpenseForm
      onSubmit={[Function]}
    />
  </div>
</div>
`;
